### PR TITLE
Touchup readme

### PR DESCRIPTION
**Pull request by Hoverbear**
After 0da3b3f5f53751c4ce5f4aeaac3a24ef34036958 I wanted to do a few more changes to our `README.md`. This mostly just tightens it up and elaborates/rewords a couple sections to be more relevant to modern PGX.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,176 +20,115 @@ This repo has relocated from `https://github.com/zombodb/pgx` to this location (
...
+- **Advanced Features**
+   + Safe access to Postgres' `MemoryContext` system via `pgx::PgMemoryContexts`
+   + Executor/planner/transaction/subtransaction hooks
+   + Safely use Postgres-provided pointers with `pgx::PgBox<T>` (akin to `alloc::boxed::Box<T>`)
+   + `#[pg_guard]` proc-macro for guarding `extern "C"` Rust functions that need to be passed into Postgres
+   + Access Postgres' logging system through `eprintln!`-like macros
+   + Direct `unsafe` access to large parts of Postgres internals via the `pgx::pg_sys` module
+   + New features added regularly!
 
 ## System Requirements
```
[the remark this answers is not in the document]
Comment (eeeebbbbrrrr) [reply README.md:64]: Sorry, the "System Requirements" sections

[the remark this answers is not in the document]
Comment (eeeebbbbrrrr) [reply README.md:64]: Screen Shot 2022-05-25 at 3 04 44 PM

[the remark this answers is not in the document]
Comment (Hoverbear) [reply README.md:64]: Should be better, I think it was the expando. I moved it lower, should be better.

**Comment by Hoverbear** [comment]
I don't want this to go stale so I'm merging it based on past feedback!

**Pull request merged**